corehq/middleware.py

```diff
@@ -210,3 +215,53 @@ def process_view(self, request, view_func, view_args, view_kwargs):
 
             if getattr(request, 'domain', None):
                 scope.set_tag('domain', request.domain)
+
+
+session_logger = logging.getLogger('session_access_log')
+
+
+def log_call(func):
+    def with_logging(*args, **kwargs):
+        session_logger.info("\n\n\n")
+        session_logger.info(func.__name__ + " was called")
+        session_logger.info("\n\n\n")
+        for line in traceback.format_stack():
+            white_list = ['corehq/', 'custom/', func.__name__]
+            if any([c in line for c in white_list]):
```

Comment:
Hey @snopoke is there a better way to discard extra lines? This one looks for lines that contain these directories or the called function name

Reply:
don't think so